src/components/FriendRequests.tsx

```diff
@@ -0,0 +1,67 @@
+'use client'
+
+import {FC, useState} from "react";
+import {Check, UserPlus, X} from 'lucide-react';
+import axios from "axios";
+import { useRouter } from 'next/navigation';
+
+interface FriendRequestsProps {
+    incomingFriendRequests: FriendRequest[]
+}
+
+type command = 'accept' | 'deny'
+
+const FriendRequests :FC<FriendRequestsProps> =({incomingFriendRequests})=>{
+    const router = useRouter()
+    const [requests, setRequests] = useState<{
+        senderId:string,
+        senderEmail:string
+    }[]>(incomingFriendRequests);
+
+    const apiPost = async (senderId: string, cmd: command) =>{
+        await axios.post(`/api/friends/${cmd}`, {id: senderId});
+        setRequests(previousOptions => previousOptions.filter(item=> item.senderId != senderId));
+        router.refresh();
+    }
+
+    const accept = async (senderId: string)=>{
+        await apiPost(senderId, 'accept')
+    }
+
+    const deny = async (senderId: string)=>{
+        await apiPost(senderId, 'deny')
+    }
+
+    return <div aria-label='friend requests'>
+        {incomingFriendRequests.length == 0 ?
+        <p className='friend-requests-nothing'>
+            Nothing to show here...
+        </p>:
+            requests.map((request)=>{
+                return (<div className='friend-requests' key={request.senderId}>
+                    <UserPlus  aria-label='add user'/>
+                    <p className='friend-requests-email'>
+                        {request.senderEmail}
+                    </p>
+                    <button aria-label={`accept friend: {r.senderEmail}`}
```

Comment:
Thing is, this component markup is going to be messy no matter what, so separate logic from presentation

Reply:
use of useState is hairy to mock. Might as well stick with the tests that are closer to the component logic for now